mlflow/docker-compose.yml

```diff
@@ -0,0 +1,6 @@
+version: "3.9"  # optional since v1.27.0
+services:
+  tracking:
+    build: ./tracking
+    args:
+        TAG: ${GITHUB_REF}
```

Comment:
Do we need this file? Can we just run `build . --build-arg TAG=${GITHUB_REF}`?

Reply:
No, the file is not necessary and was meant to simplify builds so we could use a single docker-compose build command to build all mlflow component images with the appropriate entrypoints preconfigured. However, I suppose MLFlow is a monolithic application so a single docker image could be made with the entrypoints and commands documented instead?

Reply:
I am thinking with the adjustment to the workflow, your proposed directory structure, and to easily allow other MlFlow Docker images to be made that this makes sense.

your directory structure
```
- <repo root>
  - docker
    - BaseDockerfile
    - TrackingDockerfile
    - README.md
```

BaseDockerfile
```
ARG VERSION
FROM python:3.10.6-slim-bullseye
RUN pip install --no-cache mlflow==$VERSION
```
here version comes from the build args of the workflow

TrackingDockerfile
```
ARG VERSION
FROM ghcr.io/[USERNAME]/mlflow:$VERSION
ENTRYPOINT ["mlflow", "server"]
```

Reply:
I think we can just use a single image for now.

Reply:
sounds good there is now a single dockerfile and composefile with a mini test [here](https://github.com/oojo12/mlflow/runs/8285887241?check_suite_focus=true)

going to go through and validate the other workflow today. If this gets merged before I finish that I will pull from Master to remove the compose file and update the workflow file.